emoji/src/main/java/com/gluonhq/emoji/DownloadableEmojiSpriteLoader.java

```diff
@@ -0,0 +1,176 @@
+                    }
+                    Path lockFilePath = getLockFilePath(size);
+                    LOG.fine(String.format("Creating lock file for size %s at %s", size, lockFilePath));
+                    Files.createFile(lockFilePath);
+                    LOG.fine("Download sprite file for size: " + size);
+                    String url = String.format(EMOJI_PNG_URL, commit, size);
+                    downloadFile(new URI(url).toURL(), localFilePath, lockFilePath);
+                } catch (IOException | URISyntaxException e) {
+                    LOG.severe("Download sprite failed: " + e.getMessage());
+                    throw new RuntimeException("Unable to load local image file", e);
+                }
+            }
+        }
+    }
+
+    public Image loadEmojiSprite(int size) {
+        if (!initialized) {
+            throw new RuntimeException("Sprite Loader hasn't been initialized or completed initialization");
+        }
+        try {
+            return new Image(new FileInputStream(getLocalFilePath(size).toFile()));
+        } catch (IOException e) {
+            LOG.severe("Loading of local image file failed: " + e.getMessage());
+            throw new RuntimeException("Unable to load local image file", e);
+        }
+    }
+
+    @Override
+    public InputStream loadCSV() {
+        return DownloadableEmojiSpriteLoader.class.getResourceAsStream("emoji.csv");
+    }
+
+    private void downloadFile(URL url, Path filePath, Path lockFile) throws IOException {
+        try (InputStream inputStream = url.openStream()) {
+            Files.copy(inputStream, filePath, StandardCopyOption.REPLACE_EXISTING);
+            LOG.fine("Sprite downloaded successfully as: " + filePath);
+            LOG.fine("Removing tmp file: " + lockFile);
+            Files.delete(lockFile);
+        } catch (IOException e) {
+            LOG.log(Level.SEVERE, "Downloading file failed", e);
+        }
+    }
+
+    private boolean localFileExists(int size) {
+        Path localFilePath = getLocalFilePath(size);
+        Path lockFilePath = getLockFilePath(size);
+        if (Files.exists(lockFilePath)) {
+            LOG.info("Lock file found for size: " + size);
+            LOG.info("Delete file and re-try downloading");
+            try {
+                Files.deleteIfExists(localFilePath);
+                Files.delete(lockFilePath);
+                return false;
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to delete file: " + localFilePath);
+            }
+        }
+        return Files.exists(localFilePath);
+    }
+
+    private Path getLocalFilePath(int size) {
+        String fileName = "sheet_apple_" + size + ".png";
+        return Paths.get(String.format(LOCAL_PATH, commit), fileName);
+    }
+
+    private Path getLockFilePath(int size) {
+        return Paths.get(String.format(LOCAL_PATH, commit)).resolve(size + ".downloading");
```

Comment:
does the extension `.downloading` makes sense or should I update it to `.lck` ?

Reply:
It doesn't really matter, but probably .lck or .tmp is better